File: samples/todolist-java-ebean/app/models/Task.java
package models;

import java.util.*;
import javax.persistence.*;

import play.db.ebean.*;
import play.data.format.*;
import play.data.validation.*;

@Entity 
public class Task extends Model {

    @Id
    @Constraints.Min(10)
    public Long id;
    
    @Constraints.Required
    public String name;
    
    public boolean done;
    
    @Formats.DateTime(pattern="dd/MM/yyyy")
    public Date dueDate = new Date();
    
    public static Finder<Long,Task> find = new Finder(Long.class, Task.class); 

    public static int count() {
        return new Finder(Long.class, Task.class).findRowCount();
    }

}

File: framework/play/src/test/scala/play/data/Task.scala
package play.data.models

import java.util.Date
import javax.persistence.Entity

import reflect.BeanProperty
import annotation.target.field

@Entity
class Task extends play.db.ebean.Model {

  type Id = javax.persistence.Id @field
  type Min = play.data.validation.Constraints.Min @field
  type Required = play.data.validation.Constraints.Required @field
  type DateTime = play.data.format.Formats.DateTime @field

  @Id
  @Min(10)
  @BeanProperty
  var id: Long = _

  @Required
  @BeanProperty
  var name: String = _

  @BeanProperty
  var done: Boolean = true

  @BeanProperty
  @Required
  @DateTime(pattern = "dd/MM/yyyy")
  var dueDate: Date = _

}

